cdk_emqx_cluster/cdk_emqx_cluster_stack.py

```diff
@@ -850,10 +857,33 @@ def setup_efs(self):
         self.sg_efs_mt = ec2.SecurityGroup(self, "sg_efs_mt", vpc=self.vpc)
         self.sg_efs_mt.add_ingress_rule(
             peer=self.sg, connection=ec2.Port.all_traffic())
-        # File System
-        self.shared_efs = efs.FileSystem(self, id='shared-data'+self.cluster_name, vpc=self.vpc,
-                                         removal_policy=core.RemovalPolicy.DESTROY,
-                                         lifecycle_policy=efs.LifecyclePolicy.AFTER_14_DAYS,
-                                         performance_mode=efs.PerformanceMode.GENERAL_PURPOSE,
-                                         security_group=self.sg_efs_mt
-                                         )
+
+        fsid = 'shared-data' + self.cluster_name
+
+        if self.retain_efs and self.retain_efs.startswith('fs-'):
+            # reuse existing EFS
+            fsid = self.retain_efs
+            self.shared_efs = efs.FileSystem.from_file_system_attributes(self, id=fsid, security_group=self.sg_efs_mt,
+                                                                         file_system_id=self.retain_efs)
+            # we need to explicitly add the mount targets for all private subnets
+            for net in self.vpc.private_subnets:
+                cfn_mount_target = efs.CfnMountTarget(self, 'monitoring-mountpoint',
```

Comment:
Tiny nit: looks like `cfn_mount_target` is not used?

Reply:
Cdk inside connects the EFS access point to it.